trk/src/Screens/ClimbDetail/index.js

```diff
@@ -1,32 +1,74 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import { Alert } from 'react-native';
 import TapsApi from '../../api/TapsApi';
+import ClimbsApi from '../../api/ClimbsApi';
 
 import { View, Text, StyleSheet, ScrollView, SafeAreaView, Image, TextInput, Button, TouchableWithoutFeedback, Keyboard, Share, TouchableOpacity } from 'react-native';
 
 import storage from '@react-native-firebase/storage';
 import SegmentedControl from '@react-native-segmented-control/segmented-control';
+import { AuthContext } from '../../Utils/AuthContext';
 
 
 function ClimbDetail(props) {
   console.log('[TEST] ClimbDetail called');
 
-  const { climbData } = props.route.params;
-  const { climbId } = props.route.params;
+  const { climbId, isFromHome } = props.route.params;
+  const [climbData, setClimbData] = useState(props.route.params.climbData || null);
+  const [isLoading, setIsLoading] = useState(isFromHome);
   const [climbImageUrl, setClimbImageUrl] = useState(null);
   const [setterImageUrl, setSetterImageUrl] = useState(null);
   const [completion, setCompletion] = useState('Zone');
   const [attempts, setAttempts] = useState('1');
   const [witness1, setWitness1] = useState('');
   const [witness2, setWitness2] = useState('');
+  const { currentUser } = useContext(AuthContext);
+  const [tapId, setTapId] = useState(props.route.params.tapId || null);
+
 
```

Comment:
setting the state of const that may no longer come from props.route.params (when navigating from home, climbData has not yet been read from firestore)